### PR TITLE
feat: containers, nix, appman, flatseal and redshift

### PR DESCRIPTION
**Pull request by D-Brox**
plus some small fixes and QoL improvements

### CODE REVIEW

File: src/containers.rs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        if self.enable_apptainer {
+            commands.push("sudo apt-get install apptainer");
+        }
+
+        if self.enable_qemu {
+            commands.push("sudo apt-get install qemu-system qemu-user-static qemu-utils");
+        }
+
+        if self.enable_virtualbox {
+            commands.push("sudo apt-get install virtualbox");
```
Comment (oklopfer) [inline src/containers.rs:300]: this doesn't have an `arm64` version

File: src/containers.rs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        if self.enable_podman {
+            commands.push("sudo apt-get -y docker.io");
+        }
+
+        if self.enable_podman {
+            commands.push("sudo apt-get -y podman");
+        }
+
+        if self.enable_distrobox && (self.enable_docker || self.enable_podman) {
+            commands.push("sudo PACSTALL_DOWNLOADER=quiet-wget pacstall -PI distrobox");
```
Comment (oklopfer) [inline src/containers.rs:288]: `quiet-wget` doesn't exist anymore. use the `-Q` flag

File: src/containers.rs [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        }
+
+        if self.enable_apptainer {
+            commands.push("sudo apt-get install apptainer");
```
Comment (oklopfer) [inline src/containers.rs:292]: doesnt exist in apt

File: src/containers.rs [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+                    }
+                );
+
+                self.enable_virtualbox = switched_on;
+            },
+        }
+
+        let mut commands: Vec<&str> = Vec::new();
+
+        if self.enable_podman {
```
Comment (oklopfer) [inline src/containers.rs:279]: I think this is supposed to be `enable_docker`, but this is also the wrong docker to use. `docker.io` bad and deprecated. Should be using pacstall `docker-bin` or `docker-desktop-deb` (which I need to make pacscript for)

**Review by oklopfer — CHANGES_REQUESTED** [review]

File: src/containers.rs (hunk 4 above)
Comment (oklopfer) [reply src/containers.rs:279]: also `docker-buildx-plugin-bin` and `docker-compose-plugin-bin`

`docker-desktop` has no `arm64` version either

**Comment by ajstrongdev** [comment]
To be discussed: When Flatpak (regular/beta channel) is enabled, give the user an option to install flatseal from flathub?

https://flathub.org/apps/com.github.tchx84.Flatseal - Flatseal is an application used to manage flatpak permissions.

File: src/containers.rs (hunk 4 above)
Comment (D-Brox) [reply src/containers.rs]: Make the `docker-desktop-deb` pacscript

**Comment by oklopfer** [comment]
```
pkgname="rhino-setup-git"
source=("https://github.com/D-Brox/rhino-setup.git#branch=feat/containers")
pkgver="2024.1"
pkgrel="1"
makedepends=("libgtk-4-dev" "libadwaita-1-dev" "gettext" "desktop-file-utils" "rustc" "cargo" "meson" "ninja-build")
depends=("libgtk-4-dev" "libadwaita-1-dev" "gettext" "desktop-file-utils")
gives="rhino-setup"
replaces="${gives}-bin"
pkgdesc="Rhino Linux Setup Prompt"
maintainer=("Oren Klopfer <oren@taumoda.com>")
external_connection=true
incompatible=("debian:*")

build() {
  cd "${gives}"
  meson build
  mkdir -p "${pkgdir}/etc/skel/.config/autostart"
}

package() {
  cd "${gives}"
  DESTDIR="${pkgdir}" ninja -C build install
  ln -sf "/usr/local/share/applications/org.rhinolinux.RhinoSetup.desktop" "${pkgdir}/etc/skel/.config/autostart/rhino-setup.desktop"
}

post_install() {
  mkdir -p "${homedir}/.config/autostart"
  if ! [[ -f "${homedir}/.config/autostart/rhino-setup.desktop" ]]; then
    sudo ln -sf "/usr/local/share/applications/org.rhinolinux.RhinoSetup.desktop" "${homedir}/.config/autostart/rhino-setup.desktop"
  fi
}

post_remove() {
  if [[ -L "${homedir}/.config/autostart/rhino-setup.desktop" ]]; then
    sudo rm -f "${homedir}/.config/autostart/rhino-setup.desktop"
  fi
}
```

**Review by oklopfer — APPROVED** [review]

**Pull request merged**